plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsClient.java

```diff
@@ -162,14 +181,25 @@ public List<List<Object>> readAllValues(String tableName)
         try {
             String sheetExpression = tableSheetMappingCache.getUnchecked(tableName)
                     .orElseThrow(() -> new TrinoException(SHEETS_UNKNOWN_TABLE_ERROR, "Sheet expression not found for table " + tableName));
-            return sheetDataCache.getUnchecked(sheetExpression);
+            return readAllValuesFromSheet(sheetExpression);
         }
         catch (UncheckedExecutionException e) {
             throwIfInstanceOf(e.getCause(), TrinoException.class);
             throw new TrinoException(SHEETS_TABLE_LOAD_ERROR, "Error loading data for table: " + tableName, e);
```

Comment:
`tableSheetMappingCache` on the previous line can still throw `UncheckedExecutionException`, I think this is still needed.

Reply:
you're right. it would be appropriate to pull readAllValuesFromSheet after try-catch block then

Reply:
Doing that makes the code more complex since `sheetExpression` would have to be declared outside of the try block. 

IMO I think keeping it as is makes the most sense.

Reply:
> Doing that makes the code more complex since `sheetExpression` would have to be declared outside of the try block.

correct